crates/pathfinder/src/bin/pathfinder/main.rs

```diff
@@ -114,7 +114,7 @@ async fn main() -> anyhow::Result<()> {
         state::l2::BlockValidationMode::Strict,
     ));
 
-    let shared = pathfinder_rpc::gas_price::Cached::new(Arc::new(ethereum.transport));
+    let shared = pathfinder_rpc::gas_price::Cached::new(pathfinder_context.gateway.clone());
```

Comment:
What is the point of having a standalone entity for a cached gas price, when we can pull (and update) it directly from L2 blocks during regular sync?

Reply:
The `pending` block has the most up-to-date pricing. Pending block polling is a config option and is disabled by default so most nodes won't have this information available. This would be especially incorrect when the block times are long.

I'm also averse to coupling sync and RPC even more.